CosmicsCalib/src/CaloCosmicEnergy_module.cc

```diff
@@ -0,0 +1,746 @@
+	  std::cout << "iClu: " << iClu << std::endl;
+	}
+	
+	//resetting Array
+	for(int f= 0; f< nCrystals; f++){ 
+	  PosX[f]=0;
+	  PosY[f]=0;
+	  ErrPos[f]=9.81;// 34/sqrt(12)
+	  IDs[f]=0;
+	  path[f]=0;
+	  whichHit[f]=0;
+	}
+
+	//loop over crystals
+	for(long unsigned int iCry=0 ;iCry <caloClusters[iClu].caloHitsPtrVector().size(); iCry++){
+	    
+	  //cut on energy of crystal in cluster
+	  if(caloClusters[iClu].caloHitsPtrVector()[iCry].get()->energyDep() > CutEnergyDep){
+	    //counter n crystal above 10 MeV
+	    PosX[ncry] = cal.geomUtil().mu2eToDiskFF(caloClusters[iClu].diskID(), cal.crystal(caloClusters[iClu].caloHitsPtrVector()[iCry].get()->crystalID()).position()).getX();
+	    PosY[ncry] = cal.geomUtil().mu2eToDiskFF(caloClusters[iClu].diskID(), cal.crystal(caloClusters[iClu].caloHitsPtrVector()[iCry].get()->crystalID()).position()).getY();
+	    IDs[ncry] = caloClusters[iClu].caloHitsPtrVector()[iCry].get()->crystalID();
+	    whichHit[ncry] = iCry;
+	    ncry++;
+	  }
+
+	}//end loop iCry in cluster
+	//cut on ncrystalhit
+	if(ncry > CutNCryHit){
+
+	  float min_x =PosX[0];
+	  float max_x =PosX[0]; //sbsitture xlictyposx->posX
+	  
+	  //bublbe sort to find Xmax and Xmin
+	  for(int h =1; h< ncry; h++){
+	    if(PosX[h] > max_x){
+	      max_x = PosX[h];
+	      //cout << "Max: " << max << endl;
+	    }  
+	    if(PosX[h] < min_x){
+	      min_x = PosX[h];
+	      //cout << "Min: " << min << endl;
+	    }
+	    Dx = abs(max_x-min_x);
+	  }
+
+
+	  float min_y =PosY[0];
+	  float max_y =PosY[0];
+	  
+	  //bublbe sort to find Xmax and Xmin
+	  for(int h =1; h< ncry; h++){
+	    if(PosY[h] > max_y){
+	      max_y = PosX[h];
+	      //cout << "Max: " << max << endl;
+	    }  
+	    if(PosY[h] < min_y){
+	      min_y = PosY[h];
+	      //cout << "Min: " << min << endl;
+	    }
+	    Dy = abs(max_y-min_y);
+	  }
+	  
+	  if(_diagLevel>0){
+	    std::cout << "dx: " << Dx << " dy: "<< Dy << std::endl;
+	  }
+       
+	  //geometry track
+	  TGraph *gfirst = new TGraph(ncry, PosX, PosY);
+	  TGraphErrors *gfit =  new TGraphErrors(ncry, PosX, PosY, ErrPos, ErrPos);	  
+	  gfit->SetMarkerSize(0.9);
+	  gfit->SetMarkerStyle(8);
+
+	  float chi2norm =0.;
+
+
+	  TFitResultPtr fitresult = gfirst->Fit("myPoly", "SQ");
+	  TFitResultPtr fit = gfit->Fit("myPoly", "SQ");
+	  //2 fits help convergence
+
+	  chi2norm = fit->Chi2()/fit->Ndf();
+
+
+	    //begin findpath
+
+	  float m0 = -1;
+	  m0 = myPoly->GetParameter(1);
+	  if(Dx < 33){ //if vertical tracks fit could be wrong
+	    m0 = 144.; //ca 89.6 deg
+	  }
+
+
+	  for(int h=0; h< ncry; h++){
+	    if(_diagLevel>0){   
+	      std::cout << "using m:" << m0 << " q: " << myPoly->GetParameter(0) << " x: " <<  PosX[h] << " y: "  << PosY[h]<< " ene: " << caloClusters[iClu].caloHitsPtrVector()[whichHit[h]].get()->energyDep() << std::endl;
+	    }
+	    path[h] = findpath(m0, myPoly->GetParameter(0), PosX[h], PosY[h]);
+	
+  }//end findpath
+
+	  //only verical tracks
+	  if(Dx < 34){
```

Comment:
I suggest making the numeric literals 34, 35 etc into constexpr with well chosen variable names.  This will make the code easier to read.  When you initialize the constexpr be sure to include a comment definiing the units.